-Wsign-compare warnings

<!--Please only submit issues/bug reports that come with enough information to reproduce them, ideally a unit test that fails, and possible ideas on what might be wrong. -->

<!-- Even better yet, fix the bug and/or documentation, add a unit test, and create a pull request! -->

<!-- This is a channel to report bugs/issues, not a support channel to help install/use/debug your own code. We'd love to help, but just don't have the bandwidth. Please post questions in the GTSAM Google group (https://groups.google.com/forum/#!forum/gtsam-users) -->

## Description

<!-- A clear description of the bug -->

A couple `-Wsign-compare` warnings thrown when comparing a `size_t` variable and an `int` variable.

## Steps to reproduce

1.  git clone https://github.com/borglab/gtsam
2.  mkdir build && cd build && cmake .. && make

<!-- If you have a code sample, unit test, error messages, stack traces, etc., please provide it here as well -->

```
[ 97%] Building CXX object gtsam_unstable/examples/CMakeFiles/SmartRangeExample_plaza1.dir/S    martRangeExample_plaza1.cpp.o
   /home/acxz/vcs/git/github/acxz/gtsam/gtsam_unstable/examples/TimeOfArrivalExample.cpp: In fu    nction ‘std::vector<gtsam::Event> createTrajectory(int)’:
 /home/acxz/vcs/git/github/acxz/gtsam/gtsam_unstable/examples/TimeOfArrivalExample.cpp:60:28:     warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long uns    igned int’} and ‘int’ [-Wsign-compare]
      60 |   for (size_t key = 0; key < n; key++) {                                  
         |                        ~~~~^~~                                            
   /home/acxz/vcs/git/github/acxz/gtsam/gtsam_unstable/examples/TimeOfArrivalExample.cpp: In fu    nction ‘gtsam::Values createInitialEstimate(int)’:
   /home/acxz/vcs/git/github/acxz/gtsam/gtsam_unstable/examples/TimeOfArrivalExample.cpp:117:28    : warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long un    signed int’} and ‘int’ [-Wsign-compare]
     117 |   for (size_t key = 0; key < n; key++) {                                  
         |                        ~~~~^~~          
```

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->

No `-Wsign-compare` warnings.

## Environment

<!-- Please provide information about your code environment, things such as OS, language of use (C++, Matlab or Python), version of dependent libraries if using a custom build etc., anything to provide us more information. -->

<!-- The more the information, the faster we can help resolve the issue -->

OS: ArchLinux
GCC: 9.3.0
GTSAM: latest head on develop branch at time of posting: 
b4eb0c2

## Additional information

<!-- Add any other infor or context about the problem here. -->

N/A

--- a/gtsam_unstable/examples/TimeOfArrivalExample.cpp
+++ b/gtsam_unstable/examples/TimeOfArrivalExample.cpp
@@ -53,7 +53,7 @@ vector<Point3> defineMicrophones() {
 
 /* ************************************************************************* */
 // Create ground truth trajectory
-vector<Event> createTrajectory(int n) {
+vector<Event> createTrajectory(size_t n) {
   vector<Event> trajectory;
   double timeOfEvent = 10;
   // simulate emitting a sound every second while moving on straight line
@@ -110,7 +110,7 @@ NonlinearFactorGraph createGraph(const vector<Point3>& microphones,
 
 /* ************************************************************************* */
 // create initial estimate for n events
-Values createInitialEstimate(int n) {
+Values createInitialEstimate(size_t n) {
   Values initial;
 
   Event zero;

